0.10.0 regression: UnsupportedOperationException classpath crash

From `0.10.0` on, when combining Unmock with [Detekt](https://detekt.dev/), it will fail with the following exception:

```
Caused by: java.lang.UnsupportedOperationException: Self-referencing ConfigurableFileCollections are not supported. Use the from() method to add to a ConfigurableFileCollection.
	at org.gradle.api.internal.file.collections.DefaultConfigurableFileCollection$1.startVisit(DefaultConfigurableFileCollection.java:215)

	...

	at dev.detekt.gradle.Detekt_Decorated.setProperty(Unknown Source)
	at org.gradle.internal.classpath.declarations.GroovyDynamicDispatchInterceptors.callInstrumentedSetProperty(GroovyDynamicDispatchInterceptors.java:101)
	at org.gradle.internal.classpath.declarations.GroovyDynamicDispatchInterceptors.intercept_setProperty(GroovyDynamicDispatchInterceptors.java:88)
	at de.mobilej.unmock.UnMockPlugin$_apply_closure4$_closure9.doCall$original(UnMockPlugin.groovy:72)
	at de.mobilej.unmock.UnMockPlugin$_apply_closure4$_closure9.doCall(UnMockPlugin.groovy)
```

It appears that if the classpath is a `ConfigurableFileCollection`, it does not like the `task.classpath = unmockFiles + task.classpath` syntax.

--- a/buildSrc/src/main/groovy/de/mobilej/unmock/UnMockPlugin.groovy
+++ b/buildSrc/src/main/groovy/de/mobilej/unmock/UnMockPlugin.groovy
@@ -20,6 +20,7 @@ import de.mobilej.UnMockTransform
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.api.attributes.Attribute
+import org.gradle.api.file.ConfigurableFileCollection
 import org.gradle.api.tasks.testing.Test
 
 class UnMockPlugin implements Plugin<Project> {
@@ -69,7 +70,14 @@ class UnMockPlugin implements Plugin<Project> {
         project.afterEvaluate {
             project.tasks.configureEach { task ->
                 if (task.name.contains("UnitTest") && task.hasProperty("classpath")) {
-                    task.classpath = unmockFiles + task.classpath
+                    // Some test tasks (such as when using Detekt), expose classpath as
+                    // ConfigurableFileCollection. This one does not support adding new files
+                    // with "plus". You have to use "from" method.
+                    if (task.classpath instanceof ConfigurableFileCollection) {
+                        task.classpath.from(unmockFiles)
+                    } else {
+                        task.classpath = unmockFiles + task.classpath
+                    }
                 }
             }
         }

--- a/buildSrc/src/main/groovy/de/mobilej/unmock/UnMockPlugin.groovy
+++ b/buildSrc/src/main/groovy/de/mobilej/unmock/UnMockPlugin.groovy
@@ -16,6 +16,7 @@
 
 package de.mobilej.unmock
 
+import com.android.build.gradle.tasks.factory.AndroidUnitTest
 import de.mobilej.UnMockTransform
 import org.gradle.api.Plugin
 import org.gradle.api.Project
@@ -68,17 +69,8 @@ class UnMockPlugin implements Plugin<Project> {
         def unmockFiles = project.files(unmockConfiguration)
 
         project.afterEvaluate {
-            project.tasks.configureEach { task ->
-                if (task.name.contains("UnitTest") && task.hasProperty("classpath")) {
-                    // Some test tasks (such as when using Detekt), expose classpath as
-                    // ConfigurableFileCollection. This one does not support adding new files
-                    // with "plus". You have to use "from" method.
-                    if (task.classpath instanceof ConfigurableFileCollection) {
-                        task.classpath.from(unmockFiles)
-                    } else {
-                        task.classpath = unmockFiles + task.classpath
-                    }
-                }
+            project.tasks.withType(AndroidUnitTest.class).configureEach { task ->
+                task.classpath = unmockFiles + task.classpath
             }
         }
     }

--- a/buildSrc/src/main/groovy/de/mobilej/unmock/UnMockPlugin.groovy
+++ b/buildSrc/src/main/groovy/de/mobilej/unmock/UnMockPlugin.groovy
@@ -21,8 +21,6 @@ import de.mobilej.UnMockTransform
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.api.attributes.Attribute
-import org.gradle.api.file.ConfigurableFileCollection
-import org.gradle.api.tasks.testing.Test
 
 class UnMockPlugin implements Plugin<Project> {
     void apply(Project project) {

